### PR TITLE
add sc = init_nncontext() in anomaly-detection

### PR DESCRIPTION
**Pull request by insularZJ**
to fix the following exception.
```
Py4JJavaError: An error occurred while calling o42.zooFit.
: java.lang.IllegalArgumentException: requirement failed: Engine.init: Core number is not initialized. Do you call Engine.init? See more at https://bigdl-project.github.io/master/#APIGuide/Engine/

```
@glorysdj

### CODE REVIEW

**Comment by jason-dai** [comment]
We should use init_nncontext instead of init_engine

**Comment by hkvision** [comment]
Need to add `sc = init_nncontext()` at the very beginning of this notebook

**Comment by insularZJ** [comment]
@hkvision @jason-dai

File: apps/anomaly-detection/anomaly-detection-nyc-taxi.ipynb [hunk 1]
```diff
@@ -49,6 +49,9 @@
    "source": [
     "import os\n",
     "\n",
+    "from zoo.common.nncontext import *\n"
+    "sparkConf = SparkConf().setAppName("AnomalyDetectionExample")\n"
+    "sc = init_nncontext(sparkConf)\n"
```
Comment (hkvision) [inline apps/anomaly-detection/anomaly-detection-nyc-taxi.ipynb]: can just use `sc = init_nncontext("Anomaly Detection Example")`

File: apps/anomaly-detection/anomaly-detection-nyc-taxi.ipynb [hunk 2]
```diff
@@ -49,6 +49,7 @@
    "source": [
     "import os\n",
     "\n",
+    "sc = init_nncontext(\"Anomaly Detection Example\")\n",
```
Comment (hkvision) [inline apps/anomaly-detection/anomaly-detection-nyc-taxi.ipynb]: Don't you need to import init_nncontext?

**Comment by glorysdj** [comment]
please also add other init_nncontext()
and pass the http://10.239.47.211:18888/view/ZOO-PR/job/ZOO-PR-Python-AppTests/ job

**Comment by hkvision** [comment]
Related issue: https://github.com/intel-analytics/zoo/issues/359
Some apps are not using init_nncontext.

**Comment by glorysdj** [comment]
http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Python-AppTests/156/

**Pull request merged**